src/Foundation/NSDate.cs

```diff
@@ -54,7 +63,20 @@ public partial class NSDate {
 			if (dt.Kind == DateTimeKind.Unspecified)
 				throw new ArgumentException ("DateTimeKind.Unspecified cannot be safely converted");
 
-			return FromTimeIntervalSinceReferenceDate ((dt.ToUniversalTime ().Ticks - NSDATE_TICKS) / (double) TimeSpan.TicksPerSecond);
+			var dtUnv = dt.ToUniversalTime ();
+
+			var calendar = new NSCalendar (NSCalendarType.Gregorian);
```

Comment:
Memory leak. NSCalendar is native, add a using block so that once we are done, it gets disposed.

Reply:
same, not a _real_ leak
but also a candidate to be a static member, lazily initialized on first use